config.go

```diff
@@ -107,16 +107,18 @@ func (c *Config) loadConfig() error {
 	return nil
 }
 
-func (c *Config) getNodeInfo() error {
+func (c *Config) getNodeInfo(attemptSSL bool) error {
 	log.Debug("Getting node info")
 	client, err := httpHelpers.NewMetricsClient(c.CACertificatePath, c.IAMConfigPath)
 	if err != nil {
 		return err
 	}
 
+	// If there is no available certificate, immediately drop back to HTTP
+	var useSSL = attemptSSL && len(c.IAMConfigPath) > 0
 	// Create a new DC/OS nodeutil instance
 	var stateURL = "http://leader.mesos:5050/state"
-	if len(c.IAMConfigPath) > 0 {
+	if useSSL {
 		stateURL = "https://leader.mesos:5050/state"
```

Comment:
I know that this is not part of this changeset, but it might be a good opportunity to deduplicate literals here.

```
stateURL := url.URL {
  Scheme: "http",
  Host: "leader.mesos:5050",
  Path: "/state",
}

if useSSL { 
  stateURL.Scheme = "https"
}

```
And then convert the URL to string using `stateURL.String()` below.